doc/source/tune/user-guide.rst

```diff
@@ -295,49 +295,74 @@ When running a hyperparameter search, Tune can automatically and periodically sa
  * use pre-emptible machines (by automatically restoring from last checkpoint)
  * Pausing trials when using Trial Schedulers such as HyperBand and PBT.
 
-To use Tune's checkpointing features, you must expose a ``checkpoint_dir`` argument in the function signature, and call ``tune.checkpoint_dir``:
+Let's cover how to configure your checkpoints storage location, checkpointing frequency, and how to resume from a previous run.
+
+A simple checkpointing example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 .. code-block:: python
 
-    import os
-    import time
     from ray import tune
+    from your_module import my_pytorch_training_func
+
+    # configure how your checkpoints are sync'd to the scheduler/sampler 
+    # to enable tuning. we recommend cloud storage checkpointing as it survives the
+    # cluster when your instances are terminated
+    sync_config = tune.syncConfig(
+        upload_dir="s3://my-checkpoints-bucket/path/",  # requires AWS credentials
+        sync_to_cloud=True  # this can also be a func for custom logic!
+    )
 
-    def train_func(config, checkpoint_dir=None):
-        start = 0
-        if checkpoint_dir:
-            with open(os.path.join(checkpoint_dir, "checkpoint")) as f:
-                state = json.loads(f.read())
-                start = state["step"] + 1
-
-        for step in range(start, 100):
-            time.sleep(1)
-
-            # Obtain a checkpoint directory
-            with tune.checkpoint_dir(step=step) as checkpoint_dir:
-                path = os.path.join(checkpoint_dir, "checkpoint")
-                with open(path, "w") as f:
-                    f.write(json.dumps({"step": step}))
+    tune.run(
+        # wrapping your function in tune.with_parameters() just ensures that 
+        # when checkpoints happen, all the trained parameters are saved alongside, 
+        # not just your model's hyperparameters! useful if you want your checkpoints 
+        # to be "inference ready" checkpoints of your models
+        tune.with_parameters(my_pytorch_training_func),
+
+        # name of your experiment 
+        name="my-tune-exp",
+
+        # a directory where results are stored before being
+        # sync'd to head node/cloud storage
+        local_dir="/tmp",
+
+        # see above! we will sync our checkpoints to S3 directory
+        sync_config=sync_config,
+
+        # we'll checkpoint every 10 iterations, keep the best five
+        # checkpoints (by AUC score, descending), and always checkpoint
+        # at the end of the last run!
+        checkpoint_score_attr="max-auc",
+        keep_checkpoints_num=5,
+        checkpoint_freq=10,
+        checkpoint_at_end=True,
+    )
 
-            tune.report(hello="world", ray="tune")
+In this example, checkpoints will be saved:
 
-    tune.run(train_func)
+* Locally: ``local_dir/my-tune-exp/<trial_name>/checkpoint_<step>``
```

Comment:
if sync_to_driver=False AND running locally from laptop → cluster, then this NOT stored locally

if sync_to_driver=False AND run with ray submit, then only trials run on head will be on head